vnext/Microsoft.ReactNative/Views/KeyboardEventHandler.cpp

```diff
@@ -630,14 +634,23 @@ std::string KeyboardHelper::CodeFromVirtualKey(winrt::Windows::System::VirtualKe
 bool KeyboardHelper::IsModifiedKeyPressed(
     winrt::CoreWindow const &coreWindow,
     winrt::Windows::System::VirtualKey virtualKey) {
-  return (coreWindow.GetKeyState(virtualKey) & winrt::CoreVirtualKeyStates::Down) == winrt::CoreVirtualKeyStates::Down;
+#ifdef USE_WINUI3
+  auto const &keyState = winrt::Microsoft::UI::Input::InputKeyboardSource::GetKeyStateForCurrentThread(virtualKey);
```

Comment:
Perhaps we should consider refactoring IsModifiedKeyPressed and IsModifiedKeyLocked to accept CoreVirtualKeyStates as a parameter so we can call `GetKeyStateForCurrentThread` once (the way we called `CoreWindow::GetForCurrentThread` once for UWP)

Reply:
Are you suggesting to call `GetKeyStateForCurrentThread` only once in `UpdateModifiedKeyStatusTo`? https://github.com/microsoft/react-native-windows/blob/0748834a41d71a5e3f70dfc83d0a005a83b4e8a2/vnext/Microsoft.ReactNative/Views/KeyboardEventHandler.cpp#L204-L212

Wouldn't that not work because `GetKeyStateForCurrentThread` takes in the VirtualKey as an argument and it's different for each of those calls?

Reply:
Yup, good call.